packages/ember-routing/lib/system/router.js

```diff
@@ -1109,20 +1109,21 @@ function forEachRouteAbove(originRoute, handlerInfos, callback) {
 // and are not meant to be overridable.
 let defaultActionHandlers = {
 
-  willResolveModel(transition, originRoute) {
-    originRoute.router._scheduleLoadingEvent(transition, originRoute);
+  willResolveModel(handlerInfos, transition, originRoute) {
+    this._scheduleLoadingEvent(transition, originRoute);
   },
 
   // Attempt to find an appropriate error route or substate to enter.
-  error(error, transition, originRoute) {
-    let handlerInfos = transition.state.handlerInfos;
-    let router = originRoute.router;
+  error(handlerInfos, error, transition) {
```

Comment:
FYI - The `handlerInfos` passed in here (ultimately from `router.js` microlib's `trigger` utility function) are _only_ the resolved ones, this means that `forEachRouteAbove` no longer has to iterate all of the `transition.state.handlerInfo`'s and figure out which was the culprit.

The same is true for `loading` just below...